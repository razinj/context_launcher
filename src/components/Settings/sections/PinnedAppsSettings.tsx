// React
import React from 'react'
// React Native
import { Pressable, StyleSheet, Switch, Text, View } from 'react-native'
// Components
import SettingsItemLabel from '../shared/SettingsItemLabel'
// Redux
import { useDispatch, useSelector } from 'react-redux'
import {
  clearPinnedApps,
  selectTemporaryPinnedAppsConfigMemoized,
  setTemporaryAppsConfig,
} from '../../../slices/pinnedApps'
import {
  selectDisplayPinnedAppsMemoized,
  selectDisplayTemporaryPinnedAppsMemoized,
  displayPinnedApps,
  displayTemporaryPinnedApps,
} from '../../../slices/preferences'
// Utils
import { displayToast } from '../../../utils/toast'
import { getDateFromStringWithCurrentDateValue, getTimeFromDate, stripDateFromSeconds } from '../../../utils/date'
// Contsants
import {
  activeSwitch,
  inActiveSwitch,
  settingItemButtonRippleConfig,
  settingItemWrapperStyle,
  settingsPressableItemStyle,
  switchTrackColor,
} from '../shared/values'
// TimePicker
import { DateTimePickerAndroid, DateTimePickerEvent } from '@react-native-community/datetimepicker'

const PinnedAppsSettings = () => {
  const dispatch = useDispatch()
  const displayPinnedAppsValue = useSelector(selectDisplayPinnedAppsMemoized)
  const displayTemporaryPinnedAppsValue = useSelector(selectDisplayTemporaryPinnedAppsMemoized)
  const temporaryPinnedAppsConfig = useSelector(selectTemporaryPinnedAppsConfigMemoized)

  const toggleDisplayPinnedApps = () => {
    dispatch(displayPinnedApps(!displayPinnedAppsValue))
  }

  const toggleDisplayTemporaryPinnedApps = () => {
    dispatch(displayTemporaryPinnedApps(!displayTemporaryPinnedAppsValue))
  }

  const onClearPinnedApps = () => {
    dispatch(clearPinnedApps())
    displayToast('All pinned apps cleared successfully!')
  }

  const showTimePicker = (forStartDate: boolean) => {
    DateTimePickerAndroid.open({
      mode: 'time',
      is24Hour: true,
      value: forStartDate
        ? temporaryPinnedAppsConfig.startDate
          ? getDateFromStringWithCurrentDateValue(temporaryPinnedAppsConfig.startDate)!
          : new Date()
        : temporaryPinnedAppsConfig.endDate
        ? getDateFromStringWithCurrentDateValue(temporaryPinnedAppsConfig.endDate)!
        : new Date(),
      onChange: forStartDate ? setTemporaryPinnedAppsStartDate : setTemporaryPinnedAppsEndDate,
    })
  }

  const setTemporaryPinnedAppsStartDate = (event: DateTimePickerEvent, date?: Date) => {
    if (event.type != 'set' || !date) return

    dispatch(
      setTemporaryAppsConfig({
        ...temporaryPinnedAppsConfig,
        startDate: stripDateFromSeconds(date).toString(),
      })
    )
  }

  const setTemporaryPinnedAppsEndDate = (event: DateTimePickerEvent, date?: Date) => {
    if (event.type != 'set' || !date) return

    dispatch(
      setTemporaryAppsConfig({
        ...temporaryPinnedAppsConfig,
        endDate: stripDateFromSeconds(date).toString(),
      })
    )
  }

  return (
    <>
      <View style={settingItemWrapperStyle}>
        <SettingsItemLabel title='Display pinned apps' />
        <Switch
          value={displayPinnedAppsValue}
          onValueChange={toggleDisplayPinnedApps}
          trackColor={switchTrackColor}
          thumbColor={displayPinnedAppsValue ? activeSwitch : inActiveSwitch}
        />
      </View>

      <View style={settingItemWrapperStyle}>
        <SettingsItemLabel title='Display temporary pinned apps' />
        <Switch
          value={displayTemporaryPinnedAppsValue}
          onValueChange={toggleDisplayTemporaryPinnedApps}
          trackColor={switchTrackColor}
          thumbColor={displayTemporaryPinnedAppsValue ? activeSwitch : inActiveSwitch}
        />
      </View>

      <View style={styles.pinnedAppsItemContainer}>
        <SettingsItemLabel
          title='Set start and end time of temporarliy pinned apps'
          description='Click below to set times'
        />
        <View style={styles.pinnedAppsTimePickersWrapper}>
          <Pressable
            onPress={() => showTimePicker(true)}
            android_disableSound={true}
            android_ripple={settingItemButtonRippleConfig}
            style={styles.pinnedAppsTimePicker}>
            <Text style={styles.pinnedAppsTimePickerLabel}>
              Start time is {getTimeFromDate(temporaryPinnedAppsConfig?.startDate)}
            </Text>
          </Pressable>
          <Pressable
            onPress={() => showTimePicker(false)}
            android_disableSound={true}
            android_ripple={settingItemButtonRippleConfig}
            style={styles.pinnedAppsTimePicker}>
            <Text style={styles.pinnedAppsTimePickerLabel}>
              End time is {getTimeFromDate(temporaryPinnedAppsConfig?.endDate)}
            </Text>
          </Pressable>
        </View>
      </View>

      <View style={settingItemWrapperStyle}>
        <Pressable
          onPress={onClearPinnedApps}
          android_disableSound={true}
          android_ripple={settingItemButtonRippleConfig}
          style={settingsPressableItemStyle}>
          <SettingsItemLabel title='Clear all' />
        </Pressable>
      </View>
    </>
  )
}

const styles = StyleSheet.create({
  pinnedAppsItemContainer: {
    flexDirection: 'column',
  },
  pinnedAppsTimePickersWrapper: {
    flexDirection: 'row',
    justifyContent: 'space-around',
  },
  pinnedAppsTimePicker: {
    justifyContent: 'center',
    padding: 5,
  },
  pinnedAppsTimePickerLabel: {
    color: '#808080',
  },
})

export default PinnedAppsSettings
