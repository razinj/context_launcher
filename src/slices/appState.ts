import { createAction, createSlice, PayloadAction } from '@reduxjs/toolkit'
import { AppDetails } from '../models/app-details'
import { RenderedIn } from '../models/rendered-in'
import { RootState } from '../store'

interface AppsSearchState {
  query?: string
  result: AppDetails[]
}

export interface AppState {
  displayAllApps: boolean
  displaySettings: boolean
  menuAppDetails?: AppDetails
  displayAppMenu: boolean
  displaySortableFavoriteApps: boolean
  displaySortablePinnedApps: boolean
  displaySortableTemporaryPinnedApps: boolean
  search: AppsSearchState
}

const initialState: AppState = {
  displayAllApps: false,
  displaySettings: false,
  menuAppDetails: undefined,
  displayAppMenu: false,
  displaySortableFavoriteApps: false,
  displaySortablePinnedApps: false,
  displaySortableTemporaryPinnedApps: false,
  search: {
    query: undefined,
    result: [],
  },
}

export const appStateSlice = createSlice({
  name: 'appState',
  initialState,
  reducers: {
    // GlobalAppState
    setDisplayAllApps: (state: AppState, { payload }: PayloadAction<boolean>) => {
      state.displayAllApps = payload
    },
    setDisplaySettings: (state: AppState, { payload }: PayloadAction<boolean>) => {
      state.displaySettings = payload
    },
    setMenuAppDetails: (state: AppState, { payload }: PayloadAction<AppDetails | undefined>) => {
      state.menuAppDetails = payload
    },
    setDisplayAppMenu: (state: AppState, { payload }: PayloadAction<boolean>) => {
      state.displayAppMenu = payload
    },
    setDisplaySortableFavoriteApps: (state: AppState, { payload }: PayloadAction<boolean>) => {
      state.displaySortableFavoriteApps = payload
    },
    setDisplaySortablePinnedApps: (state: AppState, { payload }: PayloadAction<boolean>) => {
      state.displaySortablePinnedApps = payload
    },
    setDisplaySortableTemporaryPinnedApps: (state: AppState, { payload }: PayloadAction<boolean>) => {
      state.displaySortableTemporaryPinnedApps = payload
    },
    // AppsSearchState
    setAppsSearchQuery: (state: AppState, { payload }: PayloadAction<string | undefined>) => {
      state.search.query = payload
    },
    setAppsSearchResult: (state: AppState, { payload }: PayloadAction<AppDetails[]>) => {
      state.search.result = payload
    },
    resetAppsSearchState: (state: AppState) => {
      state.search.query = undefined
      state.search.result = []
    },
  },
})

export const {
  setDisplayAllApps,
  setDisplaySettings,
  setMenuAppDetails,
  setDisplayAppMenu,
  setDisplaySortableFavoriteApps,
  setDisplaySortablePinnedApps,
  setDisplaySortableTemporaryPinnedApps,
  setAppsSearchQuery,
  setAppsSearchResult,
  resetAppsSearchState,
} = appStateSlice.actions

export const toogleAllApps = createAction('appState/toogleAllApps')
export const appLaunch = createAction<{ renderedIn: RenderedIn; appDetails: AppDetails }>('appState/appLaunch')
export const appLaunchFromSearch = createAction('appState/appLaunchFromSearch')
export const sortFavoriteApps = createAction('appState/sortFavoriteApps')
export const sortPinnedApps = createAction('appState/sortPinnedApps')
export const sortTemporaryPinnedApps = createAction('appState/sortTemporaryPinnedApps')

export const selectAppState = (state: RootState) => state.appState
export const selectDisplayAllApps = (state: RootState) => state.appState.displayAllApps
export const selectDisplaySettings = (state: RootState) => state.appState.displaySettings
export const selectMenuAppDetails = (state: RootState) => state.appState.menuAppDetails
export const selectDisplayAppMenu = (state: RootState) => state.appState.displayAppMenu
export const selectDisplaySortableFavoriteApps = (state: RootState) => state.appState.displaySortableFavoriteApps
export const selectDisplaySortablePinnedApps = (state: RootState) => state.appState.displaySortablePinnedApps
export const selectDisplaySortableTemporaryPinnedApps = (state: RootState) => state.appState.displaySortableTemporaryPinnedApps
export const selectAppsSearchQuery = (state: RootState) => state.appState.search.query
export const selectAppsSearchResult = (state: RootState) => state.appState.search.result

export default appStateSlice.reducer
