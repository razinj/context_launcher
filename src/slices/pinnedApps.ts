import { createSelector, createSlice, PayloadAction } from '@reduxjs/toolkit'
import { PinnedApp, PinnedAppsState, TemporaryPinnedAppsConfig } from '../models/pinned-app'
import { RootState } from '../store'
import { getAppIndex } from '../utils/apps'

const initialState: PinnedAppsState = {
  list: [],
  temporarily: [], // Temporarily pinned apps
  temporaryAppsConfig: {
    startDate: undefined,
    endDate: undefined,
  },
}

export const pinnedAppsSlice = createSlice({
  name: 'pinnedApps',
  initialState,
  reducers: {
    addPinnedApp: (state: PinnedAppsState, { payload }: PayloadAction<{ app: PinnedApp; isPermanent: boolean }>) => {
      if (payload.isPermanent) {
        const foundAppIndex = getAppIndex(state.list, payload.app.packageName)

        if (foundAppIndex !== -1) return

        state.list.push({ ...payload.app })
      } else {
        const foundAppIndex = getAppIndex(state.temporarily, payload.app.packageName)

        if (foundAppIndex !== -1) return

        state.temporarily.push({ ...payload.app })
      }
    },
    removePinnedApp: (state: PinnedAppsState, { payload }: PayloadAction<{ app: PinnedApp; isPermanent: boolean }>) => {
      if (payload.isPermanent) {
        const foundAppIndex = getAppIndex(state.list, payload.app.packageName)

        if (foundAppIndex === -1) return

        state.list.splice(foundAppIndex, 1)
      } else {
        const foundAppIndex = getAppIndex(state.temporarily, payload.app.packageName)

        if (foundAppIndex === -1) return

        state.temporarily.splice(foundAppIndex, 1)
      }
    },
    clearPinnedApps: (state: PinnedAppsState, { payload }: PayloadAction<{ temporarily: boolean }>) => {
      if (payload.temporarily) state.temporarily = []
      else state.list = []
    },
    setTemporaryAppsConfig: (state: PinnedAppsState, { payload }: PayloadAction<TemporaryPinnedAppsConfig>) => {
      state.temporaryAppsConfig = { ...payload }
    },
    setPinnedApps: (
      state: PinnedAppsState,
      { payload }: PayloadAction<{ apps: PinnedApp[]; isPermanent: boolean }>
    ) => {
      if (payload.isPermanent) state.list = payload.apps
      else state.temporarily = payload.apps
    },
  },
})

export const { addPinnedApp, removePinnedApp, setTemporaryAppsConfig, clearPinnedApps, setPinnedApps } =
  pinnedAppsSlice.actions

const selectPinnedApps = (state: RootState) => state.pinnedApps.list
const selecttTemporaryPinnedApps = (state: RootState) => state.pinnedApps.temporarily
const selectTemporaryPinnedAppsConfig = (state: RootState) => state.pinnedApps.temporaryAppsConfig

export const selectPinnedAppsMemoized = createSelector(selectPinnedApps, (list: PinnedApp[]) => list)
export const selectPinnedAppsCountMemoized = createSelector(selectPinnedApps, (list: PinnedApp[]) => list.length)
export const selectTemporaryPinnedAppsMemoized = createSelector(selecttTemporaryPinnedApps, (list: PinnedApp[]) => list)
export const selectTemporaryPinnedAppsCountMemoized = createSelector(
  selecttTemporaryPinnedApps,
  (list: PinnedApp[]) => list.length
)
export const selectTemporaryPinnedAppsConfigMemoized = createSelector(
  selectTemporaryPinnedAppsConfig,
  (config: TemporaryPinnedAppsConfig) => config
)

export default pinnedAppsSlice.reducer
