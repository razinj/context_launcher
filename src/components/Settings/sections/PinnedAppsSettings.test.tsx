import { fireEvent, screen } from '@testing-library/react-native'
import React from 'react'
import { getAppsForTests, initialStoreState } from '../../../../utils/test/data'
import { renderWithProvider } from '../../../../utils/test/utils'
import { sortPinnedApps, sortTemporaryPinnedApps } from '../../../slices/appState'
import { clearPinnedApps } from '../../../slices/pinnedApps'
import { displayPinnedApps, displayTemporaryPinnedApps } from '../../../slices/preferences'
import { RootState } from '../../../store'
import * as ToastModule from '../../../utils/toast'
import PinnedAppsSettings from './PinnedAppsSettings'

const useDispatchMock = jest.fn()

jest.mock('react-redux', () => ({
  ...jest.requireActual('react-redux'),
  useDispatch: () => useDispatchMock,
}))

describe('<PinnedAppsSettings /> Tests', () => {
  beforeAll(() => {
    jest.spyOn(ToastModule, 'displayToast')
  })

  it('should render correctly and match snapshot', () => {
    renderWithProvider(<PinnedAppsSettings />)

    expect(screen.toJSON()).toMatchSnapshot()
    expect(screen.getByTestId('display-pinned-apps-button')).toBeOnTheScreen()
    expect(screen.getByTestId('display-temporarily-pinned-apps-button')).toBeOnTheScreen()
    expect(screen.getByTestId('sort-pinned-apps-button')).toBeOnTheScreen()
    expect(screen.getByTestId('sort-temporarily-pinned-apps-button')).toBeOnTheScreen()
    expect(screen.getByTestId('set-start-time-button')).toBeOnTheScreen()
    expect(screen.getByTestId('set-end-time-button')).toBeOnTheScreen()
    expect(screen.getByTestId('clear-pinned-apps-button')).toBeOnTheScreen()
    expect(screen.getByTestId('clear-temporarily-pinned-apps-button')).toBeOnTheScreen()
  })

  it('should dispatch action to clear pinned apps when button is pressed', () => {
    renderWithProvider(<PinnedAppsSettings />)

    const clearPinnedAppsButton = screen.getByTestId('clear-pinned-apps-button')

    expect(clearPinnedAppsButton).toBeOnTheScreen()

    fireEvent.press(clearPinnedAppsButton)

    expect(useDispatchMock).toBeCalledWith(clearPinnedApps({ temporarily: false }))
    expect(ToastModule.displayToast).toBeCalledWith('Pinned apps cleared successfully!')
  })

  it('should dispatch action to clear temporarily pinned apps when button is pressed', () => {
    renderWithProvider(<PinnedAppsSettings />)

    const clearTemporarilyPinnedAppsButton = screen.getByTestId('clear-temporarily-pinned-apps-button')

    expect(clearTemporarilyPinnedAppsButton).toBeOnTheScreen()

    fireEvent.press(clearTemporarilyPinnedAppsButton)

    expect(useDispatchMock).toBeCalledWith(clearPinnedApps({ temporarily: true }))
    expect(ToastModule.displayToast).toBeCalledWith('Temporarily pinned apps cleared successfully!')
  })

  describe('Sort pinned apps button tests - permanently pinned apps', () => {
    it('should render sort pinned apps button as disabled if pinned apps are <= 1', () => {
      const customInitialStoreState: RootState = {
        ...initialStoreState,
        pinnedApps: {
          ...initialStoreState.pinnedApps,
          list: getAppsForTests(1),
        },
        preferences: {
          ...initialStoreState.preferences,
          displayPinnedApps: true,
        },
      }

      renderWithProvider(<PinnedAppsSettings />, { preloadedState: customInitialStoreState })

      expect(screen.getByTestId('sort-pinned-apps-button')).toBeDisabled()
      expect(screen.getByText(/Add more apps to be able to sort/)).toBeOnTheScreen()
    })

    it('should render sort pinned apps button as disabled if display pinned apps value is false', () => {
      const customInitialStoreState: RootState = {
        ...initialStoreState,
        pinnedApps: {
          ...initialStoreState.pinnedApps,
          list: getAppsForTests(2),
        },
        preferences: {
          ...initialStoreState.preferences,
          displayPinnedApps: false,
        },
      }

      renderWithProvider(<PinnedAppsSettings />, { preloadedState: customInitialStoreState })

      expect(screen.getByTestId('sort-pinned-apps-button')).toBeDisabled()
      expect(screen.getByText(/Display pinned apps to be able to sort/)).toBeOnTheScreen()
    })

    it('should render sort pinned apps button as enabled if pinned apps are > 1 and display pinned apps value is true', () => {
      const customInitialStoreState: RootState = {
        ...initialStoreState,
        pinnedApps: {
          ...initialStoreState.pinnedApps,
          list: getAppsForTests(2),
        },
        preferences: {
          ...initialStoreState.preferences,
          displayPinnedApps: true,
        },
      }

      renderWithProvider(<PinnedAppsSettings />, { preloadedState: customInitialStoreState })

      expect(screen.getByTestId('sort-pinned-apps-button')).not.toBeDisabled()
      expect(screen.getByText(/Click to start sorting/)).toBeOnTheScreen()
    })

    it('should dispatch action to sort pinned when button is pressed', () => {
      const customInitialStoreState: RootState = {
        ...initialStoreState,
        pinnedApps: {
          ...initialStoreState.pinnedApps,
          list: getAppsForTests(2),
        },
        preferences: {
          ...initialStoreState.preferences,
          displayPinnedApps: true,
        },
      }

      renderWithProvider(<PinnedAppsSettings />, { preloadedState: customInitialStoreState })

      const sortPinnedAppsButton = screen.getByTestId('sort-pinned-apps-button')

      expect(sortPinnedAppsButton).toBeOnTheScreen()
      expect(sortPinnedAppsButton).not.toBeDisabled()

      fireEvent.press(sortPinnedAppsButton)

      expect(useDispatchMock).toBeCalledWith(sortPinnedApps())
    })
  })

  describe('Sort pinned apps button tests - temporarily pinned apps', () => {
    it('should render sort pinned apps button as disabled if pinned apps are <= 1', () => {
      const customInitialStoreState: RootState = {
        ...initialStoreState,
        pinnedApps: {
          ...initialStoreState.pinnedApps,
          temporarily: getAppsForTests(1),
        },
        preferences: {
          ...initialStoreState.preferences,
          displayTemporaryPinnedApps: true,
        },
      }

      renderWithProvider(<PinnedAppsSettings />, { preloadedState: customInitialStoreState })

      expect(screen.getByTestId('sort-temporarily-pinned-apps-button')).toBeDisabled()
      expect(screen.getByText(/Add more apps to be able to sort/)).toBeOnTheScreen()
    })

    it('should render sort pinned apps button as disabled if display pinned apps value is false', () => {
      const customInitialStoreState: RootState = {
        ...initialStoreState,
        pinnedApps: {
          ...initialStoreState.pinnedApps,
          temporarily: getAppsForTests(2),
        },
        preferences: {
          ...initialStoreState.preferences,
          displayTemporaryPinnedApps: false,
        },
      }

      renderWithProvider(<PinnedAppsSettings />, { preloadedState: customInitialStoreState })

      expect(screen.getByTestId('sort-temporarily-pinned-apps-button')).toBeDisabled()
      expect(screen.getByText(/Display temporarily pinned apps to be able to sort/)).toBeOnTheScreen()
    })

    it('should render sort pinned apps button as enabled if pinned apps are > 1 and display pinned apps value is true', () => {
      const customInitialStoreState: RootState = {
        ...initialStoreState,
        pinnedApps: {
          ...initialStoreState.pinnedApps,
          temporarily: getAppsForTests(2),
        },
        preferences: {
          ...initialStoreState.preferences,
          displayTemporaryPinnedApps: true,
        },
      }

      renderWithProvider(<PinnedAppsSettings />, { preloadedState: customInitialStoreState })

      expect(screen.getByTestId('sort-temporarily-pinned-apps-button')).not.toBeDisabled()
      expect(screen.getByText(/Click to start sorting/)).toBeOnTheScreen()
    })

    it('should dispatch action to sort pinned when button is pressed', () => {
      const customInitialStoreState: RootState = {
        ...initialStoreState,
        pinnedApps: {
          ...initialStoreState.pinnedApps,
          temporarily: getAppsForTests(2),
        },
        preferences: {
          ...initialStoreState.preferences,
          displayTemporaryPinnedApps: true,
        },
      }

      renderWithProvider(<PinnedAppsSettings />, { preloadedState: customInitialStoreState })

      const sortTemporarilyPinnedAppsButton = screen.getByTestId('sort-temporarily-pinned-apps-button')

      expect(sortTemporarilyPinnedAppsButton).toBeOnTheScreen()
      expect(sortTemporarilyPinnedAppsButton).not.toBeDisabled()

      fireEvent.press(sortTemporarilyPinnedAppsButton)

      expect(useDispatchMock).toBeCalledWith(sortTemporaryPinnedApps())
    })
  })

  it('should dispatch action to toggle pinned apps view when button is pressed', () => {
    const customInitialStoreState: RootState = {
      ...initialStoreState,
      preferences: {
        ...initialStoreState.preferences,
        displayPinnedApps: false,
      },
    }

    renderWithProvider(<PinnedAppsSettings />, { preloadedState: customInitialStoreState })

    const togglePinnedAppsSwitch = screen.getByTestId('display-pinned-apps-button')

    expect(togglePinnedAppsSwitch).toBeOnTheScreen()
    expect(togglePinnedAppsSwitch).not.toBeDisabled()

    fireEvent(togglePinnedAppsSwitch, 'valueChange')

    expect(useDispatchMock).toBeCalledWith(displayPinnedApps(true))
  })

  it('should dispatch action to toggle temporarily pinned apps view when button is pressed', () => {
    const customInitialStoreState: RootState = {
      ...initialStoreState,
      preferences: {
        ...initialStoreState.preferences,
        displayTemporaryPinnedApps: false,
      },
    }

    renderWithProvider(<PinnedAppsSettings />, { preloadedState: customInitialStoreState })

    const toggleTemporarilyPinnedAppsSwitch = screen.getByTestId('display-temporarily-pinned-apps-button')

    expect(toggleTemporarilyPinnedAppsSwitch).toBeOnTheScreen()
    expect(toggleTemporarilyPinnedAppsSwitch).not.toBeDisabled()

    fireEvent(toggleTemporarilyPinnedAppsSwitch, 'valueChange')

    expect(useDispatchMock).toBeCalledWith(displayTemporaryPinnedApps(true))
  })

  it.todo('should open time picker with correct values - permanently pinned apps')
  it.todo('should open time picker with correct values - temporarily pinned apps')
})
