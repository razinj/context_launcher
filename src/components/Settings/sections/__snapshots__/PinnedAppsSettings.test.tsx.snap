// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PinnedAppsSettings /> Tests should render correctly and match snapshot 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    collapsable={false}
    pointerEvents="box-none"
    style={
      {
        "flex": 1,
      }
    }
  >
    <View
      style={
        {
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <View
        style={
          [
            {
              "padding": 10,
            },
            undefined,
          ]
        }
        testID="wrapper"
      >
        <Text
          style={
            [
              {
                "color": "#808080",
              },
              undefined,
            ]
          }
          testID="title"
        >
          Display pinned apps
        </Text>
      </View>
      <RCTSwitch
        accessibilityRole="switch"
        onChange={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        onTintColor="#e5e5e5"
        style={
          {
            "height": 31,
            "width": 51,
          }
        }
        testID="display-pinned-apps-button"
        thumbTintColor="#f4f3f4"
        tintColor="#e5e5e5"
        value={false}
      />
    </View>
    <View
      style={
        {
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": true,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "space-between",
            },
            {
              "opacity": 0.5,
            },
          ]
        }
        testID="sort-pinned-apps-button"
      >
        <View
          style={
            [
              {
                "padding": 10,
              },
              undefined,
            ]
          }
          testID="wrapper"
        >
          <Text
            style={
              [
                {
                  "color": "#808080",
                },
                undefined,
              ]
            }
            testID="title"
          >
            Sort pinned apps
          </Text>
          <Text
            style={
              {
                "color": "#ccc",
              }
            }
          >
            Display pinned apps to be able to sort
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        {
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <View
        style={
          [
            {
              "padding": 10,
            },
            undefined,
          ]
        }
        testID="wrapper"
      >
        <Text
          style={
            [
              {
                "color": "#808080",
              },
              undefined,
            ]
          }
          testID="title"
        >
          Display temporary pinned apps
        </Text>
      </View>
      <RCTSwitch
        accessibilityRole="switch"
        onChange={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        onTintColor="#e5e5e5"
        style={
          {
            "height": 31,
            "width": 51,
          }
        }
        testID="display-temporarily-pinned-apps-button"
        thumbTintColor="#f4f3f4"
        tintColor="#e5e5e5"
        value={false}
      />
    </View>
    <View
      style={
        {
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": true,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "space-between",
            },
            {
              "opacity": 0.5,
            },
          ]
        }
        testID="sort-temporarily-pinned-apps-button"
      >
        <View
          style={
            [
              {
                "padding": 10,
              },
              undefined,
            ]
          }
          testID="wrapper"
        >
          <Text
            style={
              [
                {
                  "color": "#808080",
                },
                undefined,
              ]
            }
            testID="title"
          >
            Sort temporarily pinned apps
          </Text>
          <Text
            style={
              {
                "color": "#ccc",
              }
            }
          >
            Display temporarily pinned apps to be able to sort
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        {
          "flexDirection": "column",
        }
      }
    >
      <View
        style={
          [
            {
              "padding": 10,
            },
            undefined,
          ]
        }
        testID="wrapper"
      >
        <Text
          style={
            [
              {
                "color": "#808080",
              },
              undefined,
            ]
          }
          testID="title"
        >
          Set start and end time of temporarliy pinned apps
        </Text>
        <Text
          style={
            {
              "color": "#ccc",
            }
          }
        >
          Click below to set times
        </Text>
      </View>
      <View
        style={
          {
            "flexDirection": "row",
            "justifyContent": "space-around",
          }
        }
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "justifyContent": "center",
              "padding": 5,
            }
          }
          testID="set-start-time-button"
        >
          <Text
            style={
              {
                "color": "#808080",
              }
            }
          >
            Start time is 
            unset
          </Text>
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "justifyContent": "center",
              "padding": 5,
            }
          }
          testID="set-end-time-button"
        >
          <Text
            style={
              {
                "color": "#808080",
              }
            }
          >
            End time is 
            unset
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        {
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }
        testID="clear-pinned-apps-button"
      >
        <View
          style={
            [
              {
                "padding": 10,
              },
              undefined,
            ]
          }
          testID="wrapper"
        >
          <Text
            style={
              [
                {
                  "color": "#808080",
                },
                undefined,
              ]
            }
            testID="title"
          >
            Clear pinned apps
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        {
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }
        testID="clear-temporarily-pinned-apps-button"
      >
        <View
          style={
            [
              {
                "padding": 10,
              },
              undefined,
            ]
          }
          testID="wrapper"
        >
          <Text
            style={
              [
                {
                  "color": "#808080",
                },
                undefined,
              ]
            }
            testID="title"
          >
            Clear temporarily pinned apps
          </Text>
        </View>
      </View>
    </View>
  </View>
</RNCSafeAreaProvider>
`;
